passagemath-ntl: build fails on Windows (MSYS2 / MINGW64)

During my attempt to package passagemath-ntl for MSYS2, I encountered the following compiler error:
```
sage/rings/polynomial/polynomial_zz_pex.cpp: In function 'PyObject* __pyx_pf_4sage_5rings_10polynomial_17polynomial_zz_pex_19Polynomial_template_2
get_cparent(__pyx_obj_4sage_5rings_10polynomial_17polynomial_zz_pex_Polynomial_template*)':
sage/rings/polynomial/polynomial_zz_pex.cpp:11303:39: error: cast from '__pyx_t_4sage_5rings_10polynomial_17polynomial_zz_pex_cparent' {aka '__pyx
_t_4sage_4libs_3ntl_16ntl_ZZ_pEContext_ZZ_pEContext_ptrs*'} to 'long int' loses precision [-fpermissive]
11303 |   __pyx_t_1 = __Pyx_PyLong_From_long(((long)__pyx_v_self->_cparent)); if (unlikely(!__pyx_t_1)) __PYX_ERR(5, 185, __pyx_L1_error)
      |                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I can work around this by just passing `-fpermissive` as additional compiler flag, but chances are this will sooner or later blow up at runtime or produce wrong values.

Another similar compiler warning (not error, so it does not cause the build to fail) I get during the passagemath-ntl build is this:
```
sage/rings/padics/pow_computer_ext.cpp: In function 'int __pyx_f_4sage_5rings_6padics_16pow_computer_ext_ZZ_pX_Eis_init(__pyx_obj_4sage_5rings_6pa
dics_16pow_computer_ext_PowComputer_ZZ_pX*, __pyx_obj_4sage_4libs_3ntl_9ntl_ZZ_pX_ntl_ZZ_pX*)':
sage/rings/padics/pow_computer_ext.cpp:5687:28: warning: right shift count >= width of type [-Wshift-count-overflow]
 5687 |     __pyx_v_D = (__pyx_v_D >> 32);
      |                  ~~~~~~~~~~^~~~~
sage/rings/padics/pow_computer_ext.cpp:5949:28: warning: right shift count >= width of type [-Wshift-count-overflow]
 5949 |     __pyx_v_D = (__pyx_v_D >> 32);
      |                  ~~~~~~~~~~^~~~~
```

My guess is the reason for both the error and the warning are different integer sizes on Windows and Linux.
On Windows (with MSVC, but also with GCC and Clang) an `int` is four bytes, `long int` is also four bytes, and `long long int` is eight bytes. On Linux it's usually four bytes for `int` and eight bytes for `long int` and `long long int`. (All assuming x86_64 CPU.) This can be seen on compiler explorer: https://godbolt.org/z/c4K3xGqnq

So some part of the code probably assumes that `long int` is always (at least) eight bytes, when it actually can be just four bytes on some platforms. The proper way to ensure that an integer type in C++ is of a certain size is to use [fixed-with integer types](https://en.cppreference.com/w/cpp/types/integer.html) like `int32_t`, `int64_t`, etc. or `int_least32_t`, `int_least64_t`, etc., available in C++11 and later.

> ```
> sage/rings/padics/pow_computer_ext.cpp: In function 'int __pyx_f_4sage_5rings_6padics_16pow_computer_ext_ZZ_pX_Eis_init(__pyx_obj_4sage_5rings_6pa
> dics_16pow_computer_ext_PowComputer_ZZ_pX*, __pyx_obj_4sage_4libs_3ntl_9ntl_ZZ_pX_ntl_ZZ_pX*)':
> sage/rings/padics/pow_computer_ext.cpp:5687:28: warning: right shift count >= width of type [-Wshift-count-overflow]
>  5687 |     __pyx_v_D = (__pyx_v_D >> 32);
>       |                  ~~~~~~~~~~^~~~~
> sage/rings/padics/pow_computer_ext.cpp:5949:28: warning: right shift count >= width of type [-Wshift-count-overflow]
>  5949 |     __pyx_v_D = (__pyx_v_D >> 32);
>       |                  ~~~~~~~~~~^~~~~
> ```

This one comes from https://github.com/passagemath/passagemath/blob/main/src/sage/rings/padics/pow_computer_ext.pyx#L101, and I think it is harmless. The condition of this branch is never true (on a system with sizeof(long) = 4); the compiler should really have eliminated it statically.

--- a/src/sage/rings/polynomial/polynomial_template.pxi
+++ b/src/sage/rings/polynomial/polynomial_template.pxi
@@ -181,9 +181,6 @@ cdef class Polynomial_template(Polynomial):
             x = parent.base_ring()(x)
             self.__class__.__init__(self, parent, x, check=check, is_gen=is_gen, construct=construct)
 
-    def get_cparent(self):
-        return <long> self._cparent
-
     def __reduce__(self):
         """
         EXAMPLES::

